packages/store-ui/src/molecules/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react'
+import { fireEvent, render } from '@testing-library/react'
+
+import Modal from './Modal'
+import Button from '../../atoms/Button'
+import Input from '../../atoms/Input'
+
+const modalTestId = 'store-modal'
+
+const TestModal = () => {
+  const [isOpen, setIsOpen] = useState(false)
+  const handleOpen = () => {
+    setIsOpen(true)
+  }
+
+  const onDismiss = () => {
+    setIsOpen(false)
+  }
+
+  return (
+    <>
+      <Button testId="trigger" onClick={handleOpen}>
+        OpenModal
+      </Button>
+      <Modal isOpen={isOpen} testId={modalTestId} onDismiss={onDismiss}>
+        <Input testId="first-input" />
+        <Button testId="first-button" />
+      </Modal>
+    </>
+  )
+}
+
+describe('Modal', () => {
+  it('Test data-store-modal', () => {
+    const { getByTestId } = render(
+      <Modal aria-label="test modal" testId="store-modal" isOpen>
+        Foo
+      </Modal>
+    )
+
+    expect(getByTestId('store-modal')).toHaveAttribute('data-store-modal')
+  })
+
+  it('Test isOpen', () => {
+    // Check that modal won't be rendered
+    const { getByTestId } = render(<TestModal />)
+
+    expect(document.querySelector(`[data-testid="${modalTestId}"]`)).toBeNull()
+
+    const modalTriggerButton = getByTestId('trigger')
+
+    fireEvent.click(modalTriggerButton)
+
+    expect(getByTestId('store-modal')).toBeInTheDocument()
+  })
+
+  // WAI-ARIA tests
+  // https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal
+  it('Test WAI-ARIA Specifications: Focus first element', () => {
```

Comment:
What about creating another test suite for those WAI-ARIA Specifications?
`describe('WAI-ARIA Specifications', () => {`

Reply:
And you could reuse some code
```
const { getByTestId } = render(<TestModal />)
const triggerModalButton = getByTestId('trigger')
```

Reply:
Done! Thank you for the suggestion, and about the second comment, I prefer to use the `getTestById('trigger')` directly.